src/Template/Element/sort_menu.ctp

```diff
@@ -0,0 +1,35 @@
+<md-menu md-offset="5 50" md-position-mode="target-right target">
+    <md-button ng-click="$mdOpenMenu($event)">
+        <md-icon>sort</md-icon> Sort by
+    </md-button>
+    <md-menu-content ng-cloak>
+
+        <?php 
+            foreach($options as $option) {
+        ?>
+
+            <md-menu-item>
+                <md-button>
+                    <md-icon>
+                        <?php 
+                            //'modified' + 'desc' parameter is not detected in show_angular and sentences of user
+                            if ($this->Paginator->sortKey() === $option['param'] && $this->Paginator->sortDir() === $option['direction']){
+                                echo 'check'; 
+                            }
+                            else {
+                                echo 'blank';
+                            }
+                        ?>
+                    </md-icon>
+                    <span class="sortOption">
+                        <?php echo $this->Paginator->sort($option['param'], $option['label'], ['direction'=>$option['direction'], 'lock'=>true]); ?>
```

Comment:
I’ve checked the documentation of the Paginator helper, but I still don’t understand the meaning of the `lock` parameter. Do you mind explaining why it’s needed?

Reply:
That was suggested to me. But I checked myself.
Basically if`locked = false`, when you click on the link, the link in reloaded page is generated in the other direction. 
So an `asc` link when clicked will be a `desc` link in new page.
It was useful with the old sort links as it was posible to manage both directions with a single link, but in the requirements I had to generate a link for each direction option.

Reply:
The old version displayed a single link (e.g. "created") and clicking it toggled the direction (that's how `Paginator->sort()` usually works).

But in this new version we display two links for each sort key (one for ascending order and one for descending order). Without `lock`, clicking on the current sort option would toggle the direction which is not what we want.

So strictly speaking `lock` is only necessary for the current sort option because all the other options still work for the first click which will reload the page.

An alternative would be to not create a link for the current sort option, i.e. clicking on it would do nothing.

Reply:
I will make a small change, I already fixed other small issues.
I think I will use the same check for the mark to find out if the link needs to be created.

Reply:
Thank you both for clarifying :pray:

Reply:
I changed the menu with new code to work using `md-button` anf `generateUrl` the `lock` option is not used anymore as url is not generated for current option-